cookbooks/bcpc-hadoop/attributes/yarn.rb

```diff
@@ -90,21 +90,29 @@
 
   site_xml['yarn.nodemanager.remote-app-log-dir'] = '/var/log/hadoop-yarn/apps'
   
-  avail_memory = node['bcpc']['hadoop']['yarn']['nodemanager']['avail_memory']
-  site_xml['yarn.nodemanager.resource.memory-mb'] =
-    avail_memory['size'] or
-    [1024, (node['memory']['total'].to_i * avail_memory['ratio']/1024).floor].max
+  yarn_max_memory = lambda do
```

Comment:
I don't know what the underlying issue is.  This was a "try it and see."  I haven't had time to revisit the problem.